charts/newrelic-infrastructure/templates/kubelet/daemonset-windows.yaml

```diff
@@ -0,0 +1,67 @@
+{{- if and .Values.kubelet.enabled .Values.kubelet.enableWindows }}
+{{- range $key, $windowsList := .Values.kubelet.windowsOsList }}
+apiVersion: apps/v1
+kind: DaemonSet
```

Comment:
this is really good. i'm slightly concerned about how there are two daemnoset templates that might have diverging settings beyond just the os-specific configurations. for example, the windows daemonset is missing `updateStrategy` that exist in the linux daemonset.

it might be worth having the two share a common daemonset template to minimize this type of divergence. if you agree with this approach, i'd recommend:

1. PR first to extract `charts/newrelic-infrastructure/templates/kubelet/daemonset.yaml` into a template like `charts/newrelic-infrastructure/templates/kubelet/_daemonset.tpl` and pass in values to recreate the existing linux daemonset. the point of this PR is to ensure the existing linux daemonset has the same output as before
2. add the windows daemonset passing different values to `_daemonset.tpl`

so the two daemonsets can share some boilerplate

Reply:
from discussion: there's a lot of dependencies between windows and linux daemonsets. so we will leave this cleanup work as follow up